src/ess/reflectometry/tools.py

```diff
@@ -270,3 +271,25 @@ def combine_curves(
         ),
         coords={'Q': qgrid},
     )
+    if any('Q_resolution' in c.coords for c in curves):
+        # This might need to be revisited. The question about how to combine curves
+        # with different Q-resolution is not completely resolved.
+        # However, in practice the difference in Q-resolution between different curves
+        # is small so it's not likely to make a big difference.
+        q_res = (
+            sc.DataArray(
+                data=c.coords.get(
+                    'Q_resolution', sc.scalar(float('nan')) * sc.values(c.data.copy())
+                ),
+                coords={'Q': c.coords['Q']},
+            )
+            for c in curves
+        )
+        qs = _interpolate_on_qgrid(q_res, qgrid).values
+        qs_avg = np.nansum(qs * inv_v, axis=0) / np.nansum(
+            ~np.isnan(qs) * inv_v, axis=0
+        )
```

Comment:
Why are we using `np.nansum` instead of `sc.nansum`?

Reply:
That's the way it's done in `scale_curves_to_overlap` (for various reasons, we're using scipy there so it's easier to just convert to numpy types once instead of going back and forth in every iteration) and this was just copied from there.

Fixed now :+1: